retrieval/targetmanager.go

```diff
@@ -309,6 +299,73 @@ func (tm *TargetManager) ApplyConfig(cfg *config.Config) bool {
 	return true
 }
 
+// prefixedTargetProvider wraps TargetProvider and prefixes source strings
+// to make source unique across a configuration.
+type prefixedTargetProvider struct {
+	TargetProvider
+
+	job       string
+	mechanism string
+	num       int
```

Comment:
For ordinal numbers (vs. cardinals), `index` or `idx` might be the clearer variable name.